src/components/NumberInput/NumberInput.stories.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { ComponentMeta, ComponentStory } from '@storybook/react';
+import { NumberInput, NumberInputProps } from './NumberInput';
+import { WithGrayBackground } from '../../hoc/WithGrayBackground';
+
+export default {
+  title: 'Components/NumberInput',
+  component: NumberInput,
+  argTypes: {
+    onChange: { action: 'changed input value' },
+  },
+} as ComponentMeta<typeof NumberInput>;
+
+const Template: ComponentStory<typeof NumberInput> = ({
+  onChange,
+  value: initialValue,
+  ...remainingProps
+}: NumberInputProps) => {
+  const [enteredValue, setEnteredValue] = useState(initialValue);
+
```

Comment:
For the story control of `initialValue` to update the value of the input on change you need to add a `useEffect()` hook that will update the `enteredValue` state to the new received `initialValue` props value on its change, e.g.
```ts
useEffect(()=>setEnteredValue(initialValue), [initialValue])
```

Reply:
Will investigate

Reply:
Refactored in pair programming session